app/models/label/LabelTable.scala

```diff
@@ -913,6 +914,52 @@ object LabelTable {
       case e: Exception => None
     }
   }
+  /**
+   * Checks if the image of the panorama exists or expired?
+   *
+   * @param gsvPanoId  Panorama ID
+   * @return           False if the image has expired.
+   */
+  def checkLabelsAndExpiration(gsvPanoId: String): Option[Boolean] = {
+    checkGoogleApiStatus(gsvPanoId) match {
+      case Some(true) =>
+        None
+      case Some(false) =>
+        GSVDataTable.markExpired(gsvPanoId, expired = true)
+        val now = new DateTime(DateTimeZone.UTC)
+        val timestamp: Timestamp = new Timestamp(now.getMillis)
+        GSVDataTable.markLastViewedForPanorama(gsvPanoId, timestamp)
+        None
+      case None => None
+    }
+  }
+  // Function to call Google API and to check the status for a given panorama ID
+
+  import scala.util.{Try, Success, Failure}
+  def checkGoogleApiStatus(gsvPanoId: String): Option[Boolean] = {
```

Comment:
Why does this look almost like an exact copy of the `panoExists()` function that we already have in this same file? We have very high standards for quality of code in this project, and we don't want to add a bunch of confusing technical debt in the future; it's not just about making something that works!

I'd like to see you take another pass over your code where you:
1. Don't repeat a bunch of code that we have elsewhere
2. Follow the conventions we use everywhere else in the code base (like imports going at the top of files)
3. Try to adhere to our [style guide](https://github.com/ProjectSidewalk/SidewalkWebpage/wiki/Style-guide)

When you submit a PR for me to review, I'm looking for it to meet all our standards, and you should therefore think that it's ready to be merged in with the rest of our code base.